### PR TITLE
[GPU] not to fuse eltwise layers that requires broadcasting

### PR DESCRIPTION
**Pull request by e-ddykim**
### Details:
 - In some cases, OneDNN gemm requires batch broadcasting for fused eltwise layers.
 - For dynamic models, we can't know the layourts for gemm and eltwise layers.
 - So, this PR prevents fusion of such eltwise layers into gemm layers.

### Tickets:
 - 146778

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp [hunk 1]
```diff
@@ -1051,6 +1051,17 @@ void prepare_primitive_fusing::fuse_simple_primitives(program &p) {
                     // this whitelist condition is temporarily and to be relaxed soon.
                     && !fused_node->is_type<fully_connected>())
                     return;
+
+                // We need to broadcast batch dimention for OneDNN binary add fusions.
+                // But, it's not possible for dynamic layers.
+                #ifdef ENABLE_ONEDNN_FOR_GPU
+                    if ((fused_node->is_type<gemm>() || fused_node->is_type<fully_connected>())
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp:1058]: Won't it prevent all gemm/fc+eltwise fusions? Looks too restrictive to me

Comment (yeonbok) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp:1058]: This is an initial draft for checking the perf impact and there was an impact. Checking another w/a to add broadcast always & skip when needed.

Comment (vladimir-paramuzov) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp:1058]: Why you just don't use unfusing feature?

Comment (e-ddykim) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp:1058]: oh... I'll also try to use the unfusing feature. Thank you!

Comment (e-ddykim) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp]: @vladimir-paramuzov I implemented it to use the unfusing feature as you suggested in https://github.com/openvinotoolkit/openvino/pull/26284. Thank you.

**Pull request closed without merging**